manim/config.py

```diff
@@ -1,326 +1,423 @@
-import argparse
-import colour
+"""
+config.py
+---------
+
+Process the manim.cfg file and the command line arguments into a single
+config object.
+
+"""
+
 import os
 import sys
-import types
-
-from .utils.tex import *
-from . import constants
-from . import dirs
+import argparse
+import configparser
+import colour
+from .utils.tex import TexTemplateFromFile, TexTemplate
 from .logger import logger
+from . import constants
+
+__all__ = ["file_writer_config", "config", "camera_config"]
+
+
+def _parse_config(config_parser, args):
+    """Parse config files and CLI arguments into a single dictionary."""
+    # By default, use the CLI section of the digested .cfg files
+    default = config_parser['CLI']
+
+    # Handle the *_quality flags.  These determine the section to read
+    # and are stored in 'camera_config'.  Note the highest resolution
+    # passed as argument will be used.
+    for flag in ['fourk_quality', 'high_quality', 'medium_quality', 'low_quality']:
+        if getattr(args, flag):
+            section = config_parser[flag]
+            break
+    else:
+        section = config_parser['CLI']
+    config = {opt: section.getint(opt) for opt in config_parser[flag]}
 
-__all__ = ["parse_cli", "get_configuration", "initialize_directories","register_tex_template","initialize_tex"]
+    # The -r, --resolution flag overrides the *_quality flags
+    if args.resolution is not None:
+        if "," in args.resolution:
+            height_str, width_str = args.resolution.split(",")
+            height, width = int(height_str), int(width_str)
+        else:
+            height, width = int(args.resolution), int(16 * height / 9)
+        config['camera_config'].update({'pixel_height': height,
+                                        'pixel_width': width})
```

Comment:
reduce amount of ops (optimize)
```suggestion
    # The -r, --resolution flag overrides the *_quality flags
    res_arg = args.resolution
    if res_arg is not None:
        if "," in res_arg:
            height_str, width_str = res_arg.split(",")
            height, width = int(height_str), int(width_str)
        else:
            height, width = int(res_arg), int(16 * height / 9)
        config['camera_config'].update({'pixel_height': height,
                                        'pixel_width': width})
```

Reply:
Once again It's unclear to me whether this is actually a _measurable_ optimization, and I think that `args.resolution` is clearer than `res_arg`.